src/sweetalert2.js

```diff
@@ -1025,17 +1027,36 @@ sweetAlert.argsToParams = (args) => {
       break
 
     case 'object':
-      showWarningsForParams(args[0])
       Object.assign(params, args[0])
       break
 
     default:
-      error('Unexpected type of argument! Expected "string" or "object", got ' + typeof args[0])
-      return false
+      throw new TypeError('Unexpected type of argument! Expected "string" or "object", got ' + typeof args[0])
   }
   return params
 }
 
+sweetAlert.plugin = function (plugin) {
+  const parentSwal = this
+  const pluginResult = plugin(parentSwal)
+  const [fn, members] = typeof pluginResult === 'function' ? [pluginResult, {}] : pluginResult
+  const childSwal = (...args) => {
+    const argsToParams = members.argsToParams || parentSwal.argsToParams
+    const params = argsToParams(args)
+    return fn(params)
+  }
+  return Object.assign(childSwal, parentSwal, members)
+}
+
+sweetAlert.mixin = function (mixin) {
```

Comment:
The same here, JSDoc with description + code example. Copy-paste from the PR description would be enough.